src/osd/OSD.cc

```diff
@@ -6706,6 +6719,9 @@ void OSD::handle_osd_map(MOSDMap *m)
   // even if this map isn't from a mon, we may have satisfied our subscription
   monc->sub_got("osdmap", last);
 
+  if (!m->maps.empty())
```

Comment:
ahh, i understand now. no, i think the current approach is better. it hides the `requested_full_first` and `requested_full_last` in *_full_map*() methods.